thrust/detail/reference.h

```diff
@@ -153,12 +153,12 @@ template<typename Element, typename Pointer, typename Derived>
     // XXX this helper exists only to avoid warnings about null references from the other swap
     template<typename System>
     inline __host__ __device__
-    void swap(System *system, derived_type &other);
+    void swap(System &system, derived_type &other);
 
     // XXX this helper exists only to avoid warnings about null references from operator value_type ()
     template<typename System>
```

Comment:
This comment indicates that this function only exists to avoid warnings about null references. Since we're no longer trafficking null pointers, this function becomes superfluous.